### PR TITLE
Migrate codebase to use ESM

### PR DESCRIPTION
**Pull request by whymarrh**
This PR migrates the majority of the JavaScript in the codebase from CommonJS (i.e. [`require`](https://nodejs.org/api/modules.html)) to ECMAScript Modules (i.e. [`import`/`export`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules)).

This has a few benefits:

1. **Consistency**—the background scripts and the UI are now more similar in style
2. **Static analysis**—all `import`s and `export`s must be defined at "compile" time instead of dynamically at runtime, allow for better static analysis and tooling (think tree-shaking, dead code elimination)
3. **Easier TypeScript migration**—this is a step towards the import syntax that TypeScript uses as TypeScript modules are aligned with ECMAScript Modules<sup>[\[1\]][1]</sup> and thus this will allows us an easier migration to TypeScript in the future

  [1]:https://www.typescriptlang.org/docs/handbook/modules.html

On that last point: I factored this task out of work I was doing to get the codebase to compile under TypeScript.

This PR is large, admittedly, and will conflict with many things. I'll be updating it as conflicts arise but hopefully we can merge this relatively quickly.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -45,7 +45,25 @@
 
   "rules": {
     "default-case": 2,
-    "import/no-unresolved": ["error", { "commonjs": true }],
+    "import/default": 2,
+    "import/export": 2,
+    "import/named": 2,
+    "import/namespace": 2,
+    "import/newline-after-import": 2,
+    "import/no-absolute-path": [2, { "commonjs": true }],
```
Comment (Gudahtt) [inline .eslintrc]: Nit: the `commonjs` option defaults to `true`, so it can be omitted.

File: babel.config.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ module.exports = function (api) {
     plugins: [
       '@babel/plugin-transform-runtime',
       '@babel/plugin-proposal-class-properties',
+      '@babel/plugin-proposal-export-default-from',
```
Comment (Gudahtt) [inline babel.config.js]: Why did you decide to add this plugin? The syntax is very nice, but it is stage 1 and doesn't seem to be required.

File: ui/app/store/store.js [hunk 3]
```diff
@@ -15,4 +15,4 @@ module.exports = function configureStore (initialState) {
       thunkMiddleware,
     ),
   ))
-}
+};
```
Comment (Gudahtt) [inline ui/app/store/store.js]: :eyes:

Comment (Gudahtt) [reply ui/app/store/store.js]: The next few files have this as well

Comment (whymarrh) [reply ui/app/store/store.js]: I'll rebase this on develop, now that #7754 is merged—that should fail this

File: babel.config.js (hunk 2 above)
Comment (whymarrh) [reply babel.config.js]: I did not know it was in Stage 1. I'll drop this and adjust the lint rules.

File: ui/app/store/store.js (hunk 3 above)
Comment (whymarrh) [reply ui/app/store/store.js]: Welp, it doesn't fail these 😢 I'll fix these manually

**Review by Gudahtt — APPROVED** [review]
Looks good!

I didn't go through this one with a fine-toothed comb, but I did look at every file, and it does seem to be mostly just the straightforward import/export conversions. The `actionConstants` change is probably the biggest thing aside from that, and I suspect ESLint would be able to catch those sort of mistakes pretty well (e.g. references to top-level exports that are now under that object).

**Review by frankiebee — APPROVED** [review]

**Review by tmashuang — APPROVED** [review]

**Pull request merged**